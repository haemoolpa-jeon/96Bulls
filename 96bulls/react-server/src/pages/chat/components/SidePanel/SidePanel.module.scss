@import "../../styles/variable";

.side-panel {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  padding: 2rem 0;
  width: 70px;
  height: 100vh;
  background-color: $color-primary-dark;
}

.icon {
  color: $color-grey-light-2;
  font-size: 44px !important;
  &:hover {
    cursor: pointer;
    opacity: 0.5;
  }
}
