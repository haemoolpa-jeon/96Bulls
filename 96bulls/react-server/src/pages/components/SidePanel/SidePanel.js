import React, { Component } from 'react';
import styles from './SidePanel.module.scss';
import CreateRoom from './CreateRoom/CreateRoom';
import firebase from '../../../config/firebase';
import PropTypes from 'prop-types';
import { clearUser, clearRoom } from '../../actions';
import { connect } from 'react-redux';
import userTypeStore from '../../../userTypeStore';

export class SidePanel extends Component {
  state = {
    CreateRoomModalIsOpen: false,
  };

  openModal = () => {
    this.setState({ CreateRoomModalIsOpen: true });
  };

  closeModal = () => {
    this.setState({ CreateRoomModalIsOpen: false });
  };

  handleLogout = () => {
    const { clearRoom, clearUser } = this.props;
    firebase
      .auth()
      .signOut()
      .then(() => console.log('Logout'));
    clearRoom();
    clearUser();
  };

  render() {
    const { CreateRoomModalIsOpen } = this.state;
    const { currentUser } = this.props;
    return (
      <div className={styles['side-panel']}>
      {userTypeStore.currentUserType === 'instructor' ?
          <button onClick={this.openModal} type="button">
          Create
      </button>
          : null}
        <button onClick={this.handleLogout} type="button">
            Logout
        </button>
        <CreateRoom
          currentUser={currentUser}
          isOpen={CreateRoomModalIsOpen}
          closeModal={this.closeModal}
        />  
      </div>
    );
  }
}

SidePanel.propTypes = {
  currentUser: PropTypes.object,
  clearRoom: PropTypes.func.isRequired,
  clearUser: PropTypes.func.isRequired,
};

export default connect(null, { clearRoom, clearUser })(SidePanel);
