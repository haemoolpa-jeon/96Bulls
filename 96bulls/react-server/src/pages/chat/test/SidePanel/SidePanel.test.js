import React from 'react';
import { SidePanel } from '../../components/SidePanel/SidePanel';
import {shallow} from 'enzyme/build';

const mockClearUser = jest.fn();
const mockClearRoom = jest.fn();

describe('SidePanel', () => {
  let component = null;
  it('초기 렌더링이 문제없이 되야함', () => {
    component = shallow(<SidePanel clearUser={mockClearUser} clearRoom={mockClearRoom} />);
  });

  it('초기 렌더링 스냅샷 일치함', () => {
    expect(component).toMatchSnapshot();
  });
});
