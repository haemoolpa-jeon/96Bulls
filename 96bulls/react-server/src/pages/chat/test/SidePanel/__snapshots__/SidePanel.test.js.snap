// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidePanel 초기 렌더링 스냅샷 일치함 1`] = `
<div
  className="side-panel"
>
  <button
    onClick={[Function]}
    type="button"
  >
    <WithStyles(Tooltip)
      title="방 만들기"
    >
      <WithStyles(Icon)
        className="icon"
      >
        add_box
      </WithStyles(Icon)>
    </WithStyles(Tooltip)>
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    <WithStyles(Tooltip)
      title="로그아웃"
    >
      <WithStyles(Icon)
        className="icon"
      >
        directions_run
      </WithStyles(Icon)>
    </WithStyles(Tooltip)>
  </button>
  <Connect(CreateRoomModal)
    closeModal={[Function]}
    isOpen={false}
  />
</div>
`;
